src/user.rs

```diff
@@ -172,38 +181,58 @@ impl UserOps for IronOxide {
         let mut rt = Runtime::new().unwrap();
         let device_create_options = device_create_options.clone();
 
-        rt.block_on(user_api::generate_device_key(
-            &recrypt,
-            &jwt.try_into()?,
-            password.try_into()?,
-            device_create_options.device_name,
-            &std::time::SystemTime::now().into(),
-            *OUR_REQUEST,
-        ))
+        rt.block_on(
+            user_api::generate_device_key(
+                &recrypt,
+                &jwt.try_into()?,
+                password.try_into()?,
+                device_create_options.device_name,
+                &std::time::SystemTime::now().into(),
+                &OUR_REQUEST,
+            )
+            .boxed()
+            .compat(),
+        )
     }
 
     fn user_delete_device(&self, device_id: Option<&DeviceId>) -> Result<DeviceId> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::device_delete(self.device.auth(), device_id))
+        rt.block_on(
+            user_api::device_delete(self.device.auth(), device_id)
+                .boxed_local() // required because something isn't Send; perhaps related to IronCoreRequest
+                .compat(),
+        )
     }
 
     fn user_verify(jwt: &str) -> Result<Option<UserResult>> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_verify(jwt.try_into()?, *OUR_REQUEST))
+        rt.block_on(
+            user_api::user_verify(jwt.try_into()?, *OUR_REQUEST)
+                .boxed()
+                .compat(),
+        )
     }
 
     fn user_get_public_key(&self, users: &[UserId]) -> Result<HashMap<UserId, PublicKey>> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_key_list(self.device.auth(), &users.to_vec()))
+        rt.block_on(
+            user_api::user_key_list(self.device.auth(), &users.to_vec())
+                .boxed()
+                .compat(),
+        )
     }
 
     fn user_rotate_private_key(&self, password: &str) -> Result<UserUpdatePrivateKeyResult> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_rotate_private_key(
-            &self.recrypt,
-            password.try_into()?,
-            self.device().auth(),
-        ))
+        rt.block_on(
+            user_api::user_rotate_private_key(
+                &self.recrypt,
+                password.try_into()?,
+                self.device().auth(),
+            )
+            .boxed_local()
```

Comment:
Why was boxed_local needed here instead of `boxed`?

Reply:
I haven't fully understood why we have to do that, yet. There are a couple of others. I added a comment here.